service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpConnectionHandler.java

```diff
@@ -76,12 +94,42 @@ protected AmqpConnectionHandler(String hostName, String userName, String sasToke
         this.sslContext = sslContext; // if null, a default SSLContext will be generated for the user
     }
 
+    protected AmqpConnectionHandler(
+            String hostName,
```

Comment:
Did we decide on using String vs url for hostname, or are we waiting for a recommendation from the sdk board on this?

Reply:
We used String for the endpoint in ADT Java, so I'm copying that same pattern here.

Reply:
the data type aside, we still expect the hostname to have a specific format right? can we add format validation?

Reply:
We do some basic non-null non-empty checks, but proton-j does the validation that this string is actually a URL for us